src/lib.rs

```diff
@@ -155,3 +157,105 @@ impl ToRepository for mongodb::Database {
         Repository::new_with_options(self.clone(), options)
     }
 }
+
+/// Contains everything you need to use MongODM.
+///
+/// It expose the underlying `mongodb` members prefixed by **Mongo** in an attempt
+/// to reduce the risk of conflict, considering the large amount of members
+/// exported by the `mongodb` driver.
+///
+/// ```rust
+/// use mongodm::prelude::*;
+///
+/// // ...
+/// ```
+pub mod prelude {
+    #[doc(no_inline)]
+    pub use crate::{ModelCursor, Repository, ToRepository, Model, CollectionConfig, SortOrder, Indexes, Index, IndexOption, sync_indexes, field, f, operator::*};
```

Comment:
Since `ToRepository` isn't a trait you'll need to implement often I suggest:
```suggestion
    pub use crate::{Repository, Model, CollectionConfig, SortOrder, Indexes, Index, IndexOption, field, f, operator::*};
    #[doc(no_inline)]
    pub use crate::ToRepository as _;
```
Also I don't think you need `ModelCursor`, `sync_indexes` that often, but maybe I'm wrong.

Reply:
I completely agree for the `ToRepository`.

`ModelCursor` ended up being used quite often in our project, for helper methods wrapping queries since the async stream api is quite verbose, and for some trait we defined for stuff like aggregation pipelines. Our use-case may or may not be representative of typical usage of this crate, however.

`sync_indexes` will most likely end up being used in a single file where initialization happen in most projects, you're right.
However, I don't find it likely to be the causes of collisions, and I kind of like the idea of exposing everything that's typically needed when working with MongODM.

I'll remove `ToRepository`, and you're free to make the decision about the 2 others, as I see your point and don't have a strong opinion on it, just a slight preference.

Reply:
I would remove `ToRepository ` too, but `ModelCursor ` & `sync_indexes ` should be kept in the prelude reexport

Reply:
To clarify `ToRepository` is not removed
```rust
use crate::ToRepository as _;
```
allow you to use the traits methods, however you can't name the trait (only useful for fully qualified syntax and trait implementation).
If you often `use` `ModelCursor` I have no objection to keep it in prelude :+1: 
In den-server I don't think we `use`d it except to build helper methods that's why I wasn't sure.
`sync_indexes` however is only used once as expected, so it's weird to put it in prelude. But anyway that's not a big deal but I can easily see someone creating his own `sync_indexes` containing all the `sync_indexes`. You would lose ability to `use` at the same time as `prelude`. That being said, this is not *that* of a big deal so if you really prefer in that way I'll not object more. Feel free to update the PR and let me know :slightly_smiling_face:

Reply:
I opted for a middleground; kept ModelCursor in the prelude, but removed sync_indexes as it is the only MongODM function that was exported by it and it's typically used only in one file.

I also fixed the futures Ext traits in the prelude with the `_` syntax, didn't know about this trick, it's pretty neat

Reply:
Great thank you! :+1: